src/it/scala/io/iohk/ethereum/sync/RegularSyncItSpec.scala

```diff
@@ -0,0 +1,70 @@
+
+import scala.concurrent.duration._
+
+class RegularSyncItSpec extends FlatSpecBase with Matchers with BeforeAndAfter {
+  implicit val testScheduler = Scheduler.fixedPool("test", 16)
+
+  it should "should sync blockchain with same best block" in customTestCaseResourceM(FakePeer.start2FakePeersRes()) {
+    case (peer1, peer2) =>
+      val blockNumer: Int = 2000
+      for {
+        _ <- peer2.importBlocksUntil(blockNumer)(IdentityUpdate)
+        _ <- peer1.connectToPeers(Set(peer2.node))
+        _ <- peer1.startRegularSync().delayExecution(50.milliseconds)
+        _ <- peer2.broadcastBlock()(IdentityUpdate).delayExecution(500.milliseconds)
+        _ <- peer1.waitForRegularSyncLoadLastBlock(blockNumer)
+      } yield {
+        assert(peer1.bl.getBestBlockNumber() == peer2.bl.getBestBlockNumber())
+      }
+  }
+
+  it should "should sync blockchain progressing forward in the same time" in customTestCaseResourceM(FakePeer.start2FakePeersRes()) {
+    case (peer1, peer2) =>
+      val blockNumer: Int = 2000
+      for {
+        _ <- peer2.startRegularSync().delayExecution(50.milliseconds)
+        _ <- peer2.importBlocksUntil(blockNumer)(IdentityUpdate)
+        _ <- peer1.connectToPeers(Set(peer2.node))
+        _ <- peer1.startRegularSync().delayExecution(500.milliseconds)
+        _ <- peer2.mineNewBlock()(IdentityUpdate).delayExecution(50.milliseconds)
+        _ <- peer1.waitForRegularSyncLoadLastBlock(blockNumer + 1)
+      } yield {
+        assert(peer1.bl.getBestBlockNumber() == peer2.bl.getBestBlockNumber())
+      }
+  }
+
+  it should "should sync peers with divergent chains will be forced to resolve branches"in customTestCaseResourceM(FakePeer.start2FakePeersRes()) {
+    case (peer1, peer2) =>
+      val blockNumer: Int = 2000
+      for {
+        _ <- peer2.importBlocksUntil(blockNumer)(IdentityUpdate)
+        _ <- peer2.startRegularSync().delayExecution(50.milliseconds)
+        _ <- peer1.importBlocksUntil(blockNumer)(IdentityUpdate)
+        _ <- peer1.startRegularSync().delayExecution(50.milliseconds)
+        _ <- peer2.mineNewBlock(10)(IdentityUpdate).delayExecution(500.milliseconds)
+        _ <- peer2.mineNewBlock(10)(IdentityUpdate).delayExecution(500.milliseconds)
+        _ <- peer2.mineNewBlock(10)(IdentityUpdate).delayExecution(500.milliseconds)
+        _ <- peer2.waitForRegularSyncLoadLastBlock(blockNumer + 3)
+        _ <- peer1.mineNewBlock()(IdentityUpdate).delayExecution(500.milliseconds)
+        _ <- peer1.waitForRegularSyncLoadLastBlock(blockNumer + 1)
+        _ <- peer1.connectToPeers(Set(peer2.node)).delayExecution(500.milliseconds)
+        _ <- peer1.waitForRegularSyncLoadLastBlock(blockNumer + 3)
+        _ <- peer2.waitForRegularSyncLoadLastBlock(blockNumer + 3)
+      } yield {
+        assert(peer1.bl.getBestBlock().number == peer2.bl.getBestBlock().number)
+        (peer1.bl.getBlockByNumber(blockNumer + 1), peer1.bl.getBlockByNumber(blockNumer + 1)) match {
```

Comment:
maybe it would be more clear to check `assert(peer1.bt.getTotalDifficultyByHash(bestBlock.hash) == peer2.bt.getTotalDifficultyByHash(bestBlock.hash))` ? i.e that peers total difficulties are the same after fork resolution

Reply:
Done